pkg/domain/infra/abi/containers.go

```diff
@@ -264,25 +264,27 @@ func (ic *ContainerEngine) ContainerRm(ctx context.Context, namesOrIds []string,
 	reports := []*entities.RmReport{}
 
 	names := namesOrIds
-	for _, cidFile := range options.CIDFiles {
-		content, err := ioutil.ReadFile(cidFile)
-		if err != nil {
-			return nil, errors.Wrap(err, "error reading CIDFile")
-		}
-		id := strings.Split(string(content), "\n")[0]
-		names = append(names, id)
-	}
-
 	// Attempt to remove named containers directly from storage, if container is defined in libpod
 	// this will fail and code will fall through to removing the container from libpod.`
 	tmpNames := []string{}
 	for _, ctr := range names {
 		report := entities.RmReport{Id: ctr}
-		if err := ic.Libpod.RemoveStorageContainer(ctr, options.Force); err != nil {
+		report.Err = ic.Libpod.RemoveStorageContainer(ctr, options.Force)
+		switch errors.Cause(report.Err) {
+		case nil:
 			// remove container names that we successfully deleted
-			tmpNames = append(tmpNames, ctr)
-		} else {
 			reports = append(reports, &report)
+		case define.ErrNoSuchCtr:
+			// remove container names that does not exist
+			reports = append(reports, &report)
```

Comment:
This seems dubious - `podman rm doesnotexist` needs to throw an error, and I'm not sure it does after this.

Reply:
Also, I think this is actually unsafe - `RemoveStorageContainer` can potentially return `ErrNoSuchCtr` in cases where the container does not exist in c/storage *but still exists in Libpod*

Reply:
Ok, I am not sure that is possible, but I can change this to fall though, since it does not change the error.

Reply:
We should not be erroring in that case, we should be proceeding to normal `RemoveContainer` *without* appending the report.

Reply:
Right it does not add the report now.